host-services/builtins/keyvalue.go

```diff
@@ -174,48 +153,34 @@ func (k *KeyValueService) handleDelete(
 	return hostservices.ServiceResultPass(200, "", resp), nil
 }
 
-func (k *KeyValueService) handleKeys(
-	nc *nats.Conn,
-	_, workload string,
-	_ []byte, _ map[string]string,
-	namespace string) (hostservices.ServiceResult, error) {
+func (k *KeyValueService) handleKeys(kvStore nats.KeyValue) (hostservices.ServiceResult, error) {
 
-	kvStore, err := k.resolveKeyValueStore(nc, namespace, workload)
-	if err != nil {
-		k.log.Warn(fmt.Sprintf("failed to resolve key/value store: %s", err.Error()))
-		return hostservices.ServiceResultFail(500, "could not resolve k/v store"), nil
-	}
-
-	// TODO: deprecated, switch to a KeysLister channel
-	keys, err := kvStore.Keys() // TODO-- paginate...
+	keyLister, err := kvStore.ListKeys()
 	if err != nil {
 		k.log.Warn(fmt.Sprintf("failed to respond to key/value host service request: %s", err.Error()))
 		return hostservices.ServiceResultFail(500, "failed to resolve keys"), nil
 	}
+	keys := make([]string, 0)
+	for key := range keyLister.Keys() {
```

Comment:
Why `range` over all the keys?

Reply:
that's not the keys, it's not a slice. that's a channel that gets closed when there are no more keys.